src/Catalyst.KBucket.Tests/ClosestTest.cs

```diff
@@ -0,0 +1,126 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Catalyst.KBucket
+{
+    /// <summary>
+    ///   From https://github.com/tristanls/k-bucket/blob/master/test/closest.js
+    /// </summary>
+    [TestClass]
+    public class ClosestTest
+    {
+        [TestMethod]
+        public void ClosestNodes()
+        {
+            var kBucket = new KBucket<Contact>();
+            for (var i = 0; i < 0x12; ++i) kBucket.Add(new Contact((byte) i));
+
+            var contact = new Contact((byte) 0x15); // 00010101
+            var contacts = kBucket.Closest(contact).Take(3).ToArray();
+            CollectionAssert.AreEqual(new byte[]
+            {
+                0x11
+            }, contacts[0].Id); // distance: 00000100
+            CollectionAssert.AreEqual(new byte[]
+            {
+                0x10
+            }, contacts[1].Id); // distance: 00000101
+            CollectionAssert.AreEqual(new byte[]
+            {
+                0x05
+            }, contacts[2].Id); // distance: 00010000
+        }
+
+        [TestMethod]
+        public void All()
+        {
+            var kBucket = new KBucket<Contact>
+            {
+                LocalContactId = new byte[] {0, 0}
+            };
+            for (var i = 0; i < 1000; ++i) kBucket.Add(new Contact((byte) (i / 256), (byte) (i % 256)));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)